source/common/ssl/ssl_socket.cc

```diff
@@ -226,6 +227,36 @@ std::string SslSocket::sha256PeerCertificateDigest() {
   return Hex::encode(computed_hash);
 }
 
+std::string SslSocket::peerCertificate() const {
+  bssl::UniquePtr<X509> cert(SSL_get_peer_certificate(ssl_.get()));
+  if (!cert) {
+    return "";
+  }
+
+  int length;
+  unsigned char *str = NULL;
+  length = ASN1_item_i2d(reinterpret_cast<ASN1_VALUE *>(cert.get()), &str, ASN1_ITEM_rptr(X509));
+  if (!str) {
+    return "";
+  }
+
+  bssl::UniquePtr<BIO> buf(BIO_new(BIO_s_mem()));
+  if(BIO_write(buf.get(), str, length) != length) {
+    return "";
+  }
+  if(!BIO_flush(buf.get())) {
+    return "";
+  }
+
+  BUF_MEM *mem = NULL;
+  BIO_get_mem_ptr(buf.get(), &mem);
+  if (mem && mem->data && mem->length) {
+    return Envoy::Base64::encode(mem->data, mem->length);
+  }
+  
+  return "";
```

Comment:
Consider rearranging the mem checks to return "" and return the actual string as the last statement:

Something like this:
BUF_MEM *mem = nullptr;
BIO_get_mem_ptr(buf.get(), &mem);
if (!mem || !mem->data || (mem->length == 0)) {
  return "";
}

return Envoy::Base64::encode(mem->data, mem->length);
}